crypto/sm2/sm2_key.c

```diff
@@ -38,7 +38,7 @@ int sm2_key_private_check(const EC_KEY *eckey)
         goto end;
     if (BN_cmp(priv_key, BN_value_one()) < 0
         || BN_cmp(priv_key, max) >= 0) {
-        ECerr(0, EC_R_INVALID_PRIVATE_KEY);
+        ERR_raise(ERR_LIB_SM2, EC_R_INVALID_PRIVATE_KEY);
```

Comment:
@levitte is this fine? (using an `EC_R_`)
Or should I create a new dedicated `SM2_R_INVALID_PRIVATE_KEY`?

Reply:
I think `SM2_R` is better...

Reply:
Pushing fixup with the new error reason in 3, 2, 1 ...